Google.Api.CommonProtos/Rpc/StatusPartial.cs

```diff
@@ -0,0 +1,132 @@
+﻿/*
+ * Copyright 2023 Google Inc. All Rights Reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+using Google.Protobuf;
+using Google.Protobuf.Reflection;
+using Google.Protobuf.WellKnownTypes;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Google.Rpc
+{
+    public partial class Status
+    {
+        /// <summary>
+        /// Cache the full names of the messages types
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        private static class MessageNameCache<T> where T : class, IMessage<T>, new()
+        {
+            public static readonly string FullName = new T().Descriptor.FullName;
```

Comment:
I know this is private to the class, but let's use a property anyway - just add `{ get; }`